src/react/reconcilation.js

```diff
@@ -286,6 +310,15 @@ export class Reconciler {
       }
     } else {
       value = this._renderFunctionalComponent(componentType, props, context);
+      if (valueIsFactroyClassComponent(this.realm, value)) {
+        invariant(value instanceof ObjectValue);
+        // TODO: use this._renderFactoryClassComponent to handle the render method (like a render prop)
+        // for now we just return the object;
+        return {
+          result: value,
```

Comment:
I don’t understand. If we don’t actually render then how can the test pass?

Reply:
We render the object. In order for us to render the internal `render` method, we need to apply a nested additionalFunction because we're inside a closure. Currently additionalFunctions only work when they are referenced to global scope. So we'll need to work on that before we can do that.

Reply:
But how does the test pass? I assume it’s supposed to compare what React really renders with what the compiler renders. If compiler doesn’t call the render method how can it pass?

Reply:
It's rendering the functional component that returns the factory object and the reconciler returns that value as the render result (the same as it was before). This passes as React handles renders that return factory objects.

Reply:
I still don’t understand. I’ll look at this when I’m at home (in an hour)

Reply:
OK. I thought `result` means "a React element" but it just means "a class or a function that we will serialize back as a type". Is that right?